### ISSUE TITLE
Question: Egress traffic from ECS Tasks

### ISSUE THREAD

**Issue by nefilim** [awslabs/aws-app-mesh-examples#74]
Hi

I have noticed that none of my ECS Tasks created with envoy & aws-appmesh-proxy-route-manager containers as per documentation are able to make any outbound TCP connection, not to the internet, nor to other resources within the same VPC:

```bash-4.4# telnet www.google.com 80
telnet: can't connect to remote host (172.217.14.228): Connection refused
bash-4.4# telnet nitroclouddevseed.cbvidc7yzeub.us-west-2.rds.amazonaws.com 80
telnet: can't connect to remote host (10.0.109.153): Connection refused

**Comment by nefilim** [comment]
Ok I see now from https://github.com/awslabs/aws-app-mesh-examples/issues/71 why specifying the RDS endpoint in `spec.backends` doesn't help, there's no matching VirtualRouter that supports that serviceName.

**Comment by bcelenza** [comment]
You should be able to model this by modeling the external site as it's own TCP VirtualNode with the DNS name in the `spec.serviceDiscovery.servceName` field. With it also defined in `spec.backends` in the downstream VirtualNode you've provided above, the destination should become available. The major caveat worth mentioning here is this will not currently work if `nitroclouddevseed.cbvidc7yzeub.us-west-2.rds.amazonaws.com` requires TLS.

This is roughly similar to the [TCP Echo VirtualNode](https://github.com/awslabs/aws-app-mesh-examples/blob/master/examples/apps/colorapp/servicemesh/config/virtualnodes/tcpecho-virtualnode.json) in our colorapp example (for which no router is defined).

**Comment by nefilim** [comment]
That's not great to be honest, are there any plans on improving this?

**Comment by bcelenza** [comment]
Yes, we are investigating ways to simplify the external passthrough use case, and TLS support is covered in #32, #33, and #34. Please upvote/comment on those issues if they are major blockers for you. We're of course looking for more customer feedback in this area.
